### PR TITLE
feat: parse embed hints from multiple stmts

### LINKED ISSUE #13: Support multi SQL hint for parser.

### PR DESCRIPTION
**Pull request by jjeffcaii**
1. add `Hints` for StmtNode
2. implement parse hints from multiple stmts

Fix #13

### CODE REVIEW

File: parser.y [hunk 1]
```diff
@@ -11109,6 +11109,18 @@ StatementList:
 			if lexer, ok := yylex.(stmtTexter); ok {
 				s.SetText(parser.lexer.client, lexer.stmtText())
 			}
+
+			// fill arana hints
+			if hints := parser.lexer.aranaHints; len(hints) > 0 {
+				type hintSetter interface {
```
Comment (cjphaha) [inline parser.y]: What do you think about this way of writing (suggestions only)?
```yacc
import (
	"strings"

	"github.com/arana-db/parser/mysql"
	"github.com/arana-db/parser/ast"
	"github.com/arana-db/parser/model"
	"github.com/arana-db/parser/opcode"
	"github.com/arana-db/parser/auth"
	"github.com/arana-db/parser/charset"
	"github.com/arana-db/parser/types"
)

type hintSetter interface {
	SetHints(hints []string)
}

%}
```

File: parser.go [hunk 2]
```diff
@@ -19042,6 +19042,18 @@ yynewstate:
 				if lexer, ok := yylex.(stmtTexter); ok {
 					s.SetText(parser.lexer.client, lexer.stmtText())
 				}
+
+				// fill arana hints
+				if hints := parser.lexer.aranaHints; len(hints) > 0 {
+					type hintSetter interface {
```
Comment (cjphaha) [inline parser.go:19048]: Can the declaration of this hintSetter interface be placed in a public place (suggestions only)?

**Review by cjphaha — APPROVED** [review]

**Review by AlexStocks — APPROVED** [review]

File: parser.y (hunk 1 above)
Comment (jjeffcaii) [reply parser.y]: 👍

File: parser.go (hunk 2 above)
Comment (jjeffcaii) [reply parser.go:19048]: 👍

**Review by Mulavar — APPROVED** [review]

**Review by binbin0325 — APPROVED** [review]

**Pull request merged**